skyportal/models.py

```diff
@@ -96,23 +98,37 @@ class Group(Base):
 
 class Stream(Base):
     name = sa.Column(sa.String, unique=True, nullable=False)
-    url = sa.Column(sa.String, unique=True, nullable=False)
-    username = sa.Column(sa.String)
-    password = sa.Column(sa.String)
+    altdata = sa.Column(JSONB, nullable=True,
+                        doc="Misc. metadata stored in JSON format, e.g. "
+                            "`{'collection': 'ZTF_alerts', selector: [1, 2]}`")
 
-    groups = relationship('Group', secondary='stream_groups',
+    groups = relationship('Group', secondary='group_streams',
                           back_populates='streams',
                           passive_deletes=True)
+    users = relationship('User', secondary='stream_users',
+                         back_populates='streams',
+                         passive_deletes=True)
+    filters = relationship('Filter',
+                           back_populates='stream',
+                           passive_deletes=True)
+
 
+GroupStream = join_model('group_streams', Group, Stream)
```

Comment:
If a stream is deleted, will it delete all the associated users in that stream? We should ensure that doesn't happen...

Reply:
This will not happen, `test_add_stream_to_single_user_group_delete_stream` in `test_groups.py` tests for that.